rclpy/rclpy/task.py

```diff
@@ -42,6 +44,12 @@ def __init__(self, *, executor=None):
         self._executor = None
         self._set_executor(executor)
 
+    def __del__(self):
+        if self._exception is not None and self._exception_fetched:
```

Comment:
`and not self._exception_fetched`?

Reply:
Very good point. I changed the logic to match the naming of the 
variable: c63e3be660521b8473cc6e667212669da0c8044f